src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -302,16 +331,19 @@ private unsafe bool HandleItems(ActionType actionType, QueryOperationContext que
                 }
             }
 
+            _index.LastProcessedReference.Clear(actionType);
+
             return moreWorkFound;
         }
 
         private IEnumerable<IndexItem> GetItemsFromCollectionThatReference(QueryOperationContext queryContext, TransactionOperationContext indexContext,
-            string collection, Reference referencedDocument, long lastIndexedEtag, HashSet<string> indexed)
+            string collection, Reference referencedDocument, long lastIndexedEtag, HashSet<string> indexed, ActionType actionType)
         {
-            foreach (var key in _referencesStorage.GetItemKeysFromCollectionThatReference(collection, referencedDocument.Key, indexContext.Transaction))
+            //TODO: support Counters/TimeSeries/CompareExchange indexes references as well
```

Comment:
Under what scenario can we have a _lot_ of values for a counter / time series? That is effectively well bounded, no?

A single compare exchange may be used by many documents, though.

Reply:
There can be and I don't take that into account (if we the same id for compare exchange and a document for example)